pkg/analyzer/nodes/node_test.go

```diff
@@ -0,0 +1,310 @@
+			args: args{
+				fileMatcherMock: matcherMock{
+					matchFunc: func(_ string, _ os.FileInfo, _ io.ReaderAt) (bool, bool) {
+						return true, true
+					},
+				},
+				entryMock: dirEntryMock{infoFunc: func() (fs.FileInfo, error) {
+					ff := analyzertest.NewFakeFile("foobar", []byte("foobar is too big"), 0644)
+					return ff.FileInfo(), nil
+				}},
+			},
+			wantFileData: nil,
+			wantErr:      assert.NoError,
+		},
+		{
+			name: "when file is extractable then return it",
+			args: args{
+				fileMatcherMock: matcherMock{
+					matchFunc: func(_ string, _ os.FileInfo, _ io.ReaderAt) (bool, bool) {
+						return true, true
+					},
+				},
+				entryMock: dirEntryMock{infoFunc: func() (fs.FileInfo, error) {
+					ff := analyzertest.NewFakeFile("foobar", []byte{}, 0755)
+					return ff.FileInfo(), nil
+				}},
+			},
+			wantFileData: &fileMetadata{
+				isExecutable:  true,
+				isExtractable: true,
+			},
+			wantErr: assert.NoError,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.maxFileSize > 0 {
+				prevMaxFileSize := analyzer.GetMaxExtractableFileSize()
+				analyzer.SetMaxExtractableFileSize(tt.maxFileSize)
+				defer func() {
+					analyzer.SetMaxExtractableFileSize(prevMaxFileSize)
+				}()
+			}
+			got, err := extractFile(tt.args.path, tt.args.entryMock, tt.args.fileMatcherMock, nil)
+			if !tt.wantErr(t, err) {
+				return
+			}
+			assert.Equal(t, tt.wantFileData, got)
+		})
+	}
+}
+
+func Test_filesMap_Get(t *testing.T) {
+	wd, _ := os.Getwd()
+	testdata := filepath.Join(wd, "testdata")
+	type fields struct {
+		root      string
+		fileMap   map[string]*fileMetadata
+		readError error
+	}
+	tests := []struct {
+		name         string
+		fields       fields
+		path         string
+		wantFileData analyzer.FileData
+		wantExists   bool
+		wantError    error
+	}{
+		{
+			name: "when path not in map then returns does not exist",
+			path: "whatever",
+		},
+		{
+			name: "when file is not extractable then returns without content",
+			fields: fields{
+				root: "",
+				fileMap: map[string]*fileMetadata{
+					"foo/bar.txt": {
+						isExecutable:  true,
+						isExtractable: false,
+					},
+				},
+				readError: nil,
+			},
+			path: "foo/bar.txt",
+			wantFileData: analyzer.FileData{
+				Executable: true,
+			},
+			wantExists: true,
+		},
+		{
+			name: "when file is in map and is extractable, should return with contents",
+			fields: fields{
+				root: filepath.Join(testdata, "rootfs-foo"),
+				fileMap: map[string]*fileMetadata{
+					"etc/redhat-release": {
+						isExecutable:  false,
+						isExtractable: true,
+					},
+				},
+				readError: nil,
+			},
+			path: "etc/redhat-release",
+			wantFileData: analyzer.FileData{
+				Contents: []byte("Some random red hat release that does not exist (X.Y) (foo bar)\n"),
+			},
+			wantExists: true,
+		},
+		{
+			name: "when file is in map and is extractable but does not exist, then keep error",
```

Comment:
how does this happen?

Reply:
Between extraction and reading the file might have moved, was deleted, changed permission or something else happened to the underlying filesystem (e.g. failures).